helm/nats-server/templates/configmap.yaml

```diff
@@ -0,0 +1,109 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "nats.name" . }}-config
+  labels:
+    app: {{ template "nats.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+data:
+  nats.conf: |
+    # PID file shared with configuration reloader.
+    pid_file: "/var/run/nats/nats.pid"
+
+    ###############
+    #             #
+    # Monitoring  #
+    #             #
+    ###############
+    http: 8222
+    server_name: $POD_NAME
+
+    {{ if .Values.cluster.enabled }}
+    ###################################
+    #                                 #
+    # NATS Full Mesh Clustering Setup #
+    #                                 #
+    ###################################
+    cluster {
+      port: 6222
+
+      routes = [
+        {{ template "nats.clusterRoutes" . }}
+      ]
+      cluster_advertise: $CLUSTER_ADVERTISE
+
+      connect_retries: {{ .Values.nats.connectRetries }}
+    }
+    {{ end }}
+
+    {{- if and .Values.nats.advertise .Values.nats.externalAccess }}
+    include "advertise/client_advertise.conf"
+    {{- end }}
+
+    {{- if .Values.leafnodes.enabled }}
+    leafnodes {
+      listen: "0.0.0.0:7422"
+      {{ if and .Values.nats.advertise .Values.nats.externalAccess }}
+      include "advertise/gateway_advertise.conf"
+      {{ end }}
+    }
+    {{ end }}
+
+    ####################
+    #                  #
+    # Logging Options  #
+    #                  #
+    ####################
+    {{- with .Values.nats.logging.debug }}
+    debug: {{ . }}
+    {{- end }}
```

Comment:
What's the reason for using the `with` operator here instead of just referencing the value directly?

Reply:
these templates are largely copies from nats.io noting they have not yet pushed their helm-charts to helm.sh and there are some refinements we would require (namely ability to vary the namespace) that they have agreed to, a pr just needs to be filed.

Reply:
the template author chose to use the with flow-control  option, perhaps to narrow scope and inline the entire .Values.nats.logging.debug block if/when it exists.

see: https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with